superset/connectors/sqla/models.py

```diff
@@ -1427,7 +1428,19 @@ def _get_top_groups(  # pylint: disable=no-self-use
 
         return or_(*groups)
 
+    def fill_column_types(self) -> str:
+        for column in self.columns:
+            self.column_types[column.column_name] = column.type
+
     def query(self, query_obj: QueryObjectDict) -> QueryResult:
+        if self.column_types == {}:
+            self.fill_column_types()
+
+        # Converts comparator from string to boolean if field type is boolean
+        for i in range(len(query_obj['filter'])):
+            if self.column_types[query_obj['filter'][i]['col']] == 'BOOLEAN':
+                query_obj['filter'][i]['val'] = bool(query_obj['filter'][i]['val'].lower())
+
```

Comment:
Im still not sure how we want to go about this. Tom mentioned maybe tackling this at the Trino connector level. So I will revisit this once we discuss that